### PR TITLE
Fix bug in small allocator in case of small slab_alloc_factor

### LINKED ISSUE #5216: Strange behaviour of small allocator
Tarantool version: any

The code

```lua
box.cfg{wal_mode='none', memtx_memory=2*1024*1024*1024, listen=3301, memtx_min_tuple_size=32, slab_alloc_factor=1.03}

s = box.schema.space.create('test')
_ = s:create_index('test')

function str(i) return string.rep('a', math.floor(256 * math.pow(1.03, i))) end
for i=1,276 do s:replace{i+200, str(i)} end
for i=1,274 do s:delete{i+200} end
collectgarbage('collect')
s:replace{200+277, str(275)}
s:delete{200+275}
collectgarbage('collect')
s:delete{200+276}
collectgarbage('collect')
```

breaks debug version of tarantool:

```
tarantool: /home/alyapunov/Work/tarantool/src/lib/small/small/small.c:291: mempool_find: Assertion `size >= upper_bound->objsize_min' failed.
Aborted (core dumped)
```
and puts release version to unexpected state (see mem_used):

```
tarantool> box.slab.stats()[#box.slab.stats()-2]
---
- mem_free: 17654224
  mem_used: 18446744073708674504
  item_count: 3069965951
  item_size: 877112
  slab_count: 1
  slab_size: 16777216
...
```

### PR DESCRIPTION
**Pull request by mechanik20051988**
Change small allocator behavior and add test for #5216 issue

Closes #5216

### CODE REVIEW

**Review by alyapunov — APPROVED** [review]
LGTM

File: src/box/memtx_engine.c [hunk 1]
```diff
@@ -1104,6 +1104,14 @@ memtx_engine_new(const char *snap_dirname, bool force_recovery,
 	if (objsize_min < OBJSIZE_MIN)
 		objsize_min = OBJSIZE_MIN;
 
+	if (alloc_factor > 2) {
+		say_error("Alloc factor must be less than or equal to 2.0. It will be reduced to 2.0");
+		alloc_factor = 2.0;
+	} else if (alloc_factor <= 1.0) {
+		say_error("Alloc factor must be grater than then 1.0. It will be increased to 1.001");
```
Comment (Totktonada) [inline src/box/memtx_engine.c:1111]: Typo: grater → greater.

**Comment by Totktonada** [comment]
Please, beware of https://github.com/tarantool/small/issues/20. We should push the small changes to `master` (and `tarantool-x.y` if necessary) first and only then promote the submodule in tarantool's `master` (and `x.y` branches if necessary).

I remind about this just in case: many times I saw the process was not done right.

File: src/box/memtx_engine.c (hunk 1 above)
Comment (mechanik20051988) [reply src/box/memtx_engine.c]: yes thanks, changed by force pushed

**Pull request merged**